src/builder/commands/generic.rs

```diff
@@ -303,19 +300,18 @@ impl BuildStep for RunAs {
         for i in self.supplementary_gids.iter() {
             hash.text("supplementary_gids", &i);
         }
-        hash.field("RunAs", &self.script);
+        hash.field("script", &self.script);
         Ok(())
     }
     fn build(&self, guard: &mut Guard, build: bool)
         -> Result<(), StepError>
     {
         if build {
-            let mut cmd = try!(build_command(&mut guard.ctx,
-                                             &["/bin/sh".to_string(),
-                                               "-exc".to_string(),
-                                               self.script.to_string()],
-                                             &Path::new("/work"),
-                                             &[]));
+            let mut cmd = Command::new(&"/bin/sh".to_string());
```

Comment:
No need for `to_string()` as far as I remember. That's how magic works in `File::open` and similar things.